### PR TITLE
Update instructions for building from source

### LINKED ISSUE #1640: [1pt] feedback: Building from source
> <…>n-msgpack python-yaml python-argparse python-six python-gevent$ **|yum install -y gcc gcc-c++ cmake make coreutils sed \
>       autoconf automake libtool zlib-devel \
>       readline-devel ncurses-devel openssl-devel \
>       libunwind-devel libicu-devel \
>       python python-pip python-setuptools python-devel \
>       python-msgpack python-yaml python-argparse python-six python-gevent|**
> 
> 
> 
> For CentOS 8, say:
> COPYyum install epel-release
> curl -s http<…>

https://www.tarantool.io/en/doc/2.2/dev_guide/building_from_source/

Please add "git" package to this snippet

### LINKED ISSUE #1702: [2pt] mopinion: mistake in building instruction
URL
https://www.tarantool.io/en/doc/latest/dev_guide/building_from_source/

'For CentOS 8, say...' fails on sed and coreutils

### PR DESCRIPTION
**Pull request by Onvember**
Fixes #1702
Fixes #1640

### CODE REVIEW

**Comment by NickVolynkin** [comment]
@Totktonada I've got several concerns:

1. We don't freeze package versions here, so build environment isn't controlled. Someone can follow the instructions exactly and still face a problem. Moreover, we recommend exact versions for *some* packages in the instructions, but then don't repeat them in the code snippet.
2. Package lists and build instructions have no connection with our tests for the corresponding platforms. We will have to update instructions, and we will forget to do so.
3. Package lists and build instructions aren't in the form of code.

File: doc/dev_guide/building_from_source.rst [hunk 1]
```diff
@@ -112,7 +112,7 @@ the preferences can differ. But strategically the steps are always the same.
                autoconf automake libtool \
                readline ncurses openssl libunwind icu \
                python27 py27-pip py27-setuptools py27-daemon \
-               py27-msgpack py27-yaml py27-argparse py27-six py27-gevent
+               py27-msgpack py27-yaml py27-six py27-gevent
```
Comment (NickVolynkin) [inline doc/dev_guide/building_from_source.rst:115]: Do we really use Python2? @Totktonada

Comment (Onvember) [reply doc/dev_guide/building_from_source.rst:115]: As I can see, related issues are still open so I decided lo leave it as is (https://github.com/tarantool/doc/issues/1757)

Comment (Totktonada) [reply doc/dev_guide/building_from_source.rst:115]: It is so for now. Should be resolved within two weeks I guess.

**Comment by Totktonada** [comment]
> 1. We don't freeze package versions here, so build environment isn't controlled. Someone can follow the instructions exactly and still face a problem. Moreover, we recommend exact versions for some packages in the instructions, but then don't repeat them in the code snippet.

Packages are installed from the distribution specific repositories. Any binary distribution must ensure binary compatibility of packages within a distribution version. A user should not meet any problems related to packages version on a distribution we support.

I don't see frozen versions or recommendation to use it aside of PyYAML (the mentioned version is quite old anyway). I don't know why the instruction repeats `test-run/requirements.txt`. It should not.

**Comment by Totktonada** [comment]
> 2. Package lists and build instructions have no connection with our tests for the corresponding platforms. We will have to update instructions, and we will forget to do so.

Do you propose to extract dependencies from `rpm/tarantool.spec` or `debian/control`? That's the good idea, but requires some research. Consider [this](https://github.com/packpack/packpack/blob/c1d29ef09c6a89a89358ef56d7eca12843490cc5/pack/rpm.mk#L164-L165) and [this](https://github.com/packpack/packpack/blob/c1d29ef09c6a89a89358ef56d7eca12843490cc5/pack/deb.mk#L133).

**Comment by Totktonada** [comment]
> 3. Package lists and build instructions aren't in the form of code.

Is that about formatting on the web page?

File: doc/dev_guide/building_from_source.rst [hunk 2]
```diff
@@ -38,43 +38,43 @@ the preferences can differ. But strategically the steps are always the same.
...
-       * `pyyaml <https://pypi.python.org/pypi/PyYAML>`_ version 3.10
-       * `argparse <https://pypi.python.org/pypi/argparse>`_ version 1.1
-       * `msgpack-python <https://pypi.python.org/pypi/msgpack-python>`_ version 0.4.6
-       * `gevent <https://pypi.python.org/pypi/gevent>`_ version 1.1.2
-       * `six <https://pypi.python.org/pypi/six>`_ version 1.8.0
+     * `pyyaml <https://pypi.python.org/pypi/PyYAML>`_ version 3.10
+     * `argparse <https://pypi.python.org/pypi/argparse>`_ version 1.1
+     * `msgpack-python <https://pypi.python.org/pypi/msgpack-python>`_ version 0.4.6
+     * `gevent <https://pypi.python.org/pypi/gevent>`_ version 1.1.2
+     * `six <https://pypi.python.org/pypi/six>`_ version 1.8.0
```
Comment (Totktonada) [inline doc/dev_guide/building_from_source.rst:45]: * argparse — it is built-in in Python 2.6+ AFAIR.
* msgpack-python is a submdoule of test-run, so it is not necessary to install it separately.
* All other versions seems obsoleted. And there is not any sense to list them, they're listed in `test-run/requirements.txt` in the format that pip understands. One who want to install it using a package manager can find related package names in `rpm/tarantol.spec` or `debian/control` (but I would left this exercise to a user, just mention where to find it).

File: doc/dev_guide/building_from_source.rst [hunk 3]
```diff
@@ -38,43 +38,43 @@ the preferences can differ. But strategically the steps are always the same.
...
      .. code-block:: console
 
-         $ yum install -y gcc gcc-c++ cmake make coreutils sed \
+         $ yum install -y gcc gcc-c++ git cmake make coreutils sed \
                autoconf automake libtool zlib-devel \
                readline-devel ncurses-devel openssl-devel \
                libunwind-devel libicu-devel \
                python python-pip python-setuptools python-devel \
-               python-msgpack python-yaml python-argparse python-six python-gevent
+               python-msgpack python-yaml python-six python-gevent
```
Comment (Totktonada) [inline doc/dev_guide/building_from_source.rst:69]: That's is not so trivial considering that it is Python 2 packages on CentOS 7, but Python 3 ones on CentOS 8 / Fedora. I would just suggest to use `pip --user -r test-run/requirements.txt` or `pip -r test-run/requirements.txt` from a virtualenv environments. We can mention that the same can be installed with packages, but, to be honest, if one want, (s)he will do it anyway based on the requirements.txt content. So, let's stick to `pip` for simplicity.

**Comment by Totktonada** [comment]
Please, also eliminate mentions of 2.2 branch. It is not supported for a long time and useless for a user in fact.

**Comment by Totktonada** [comment]
> 5. Run the test suite.

This section is something really strance. You should just have all dependencies available, build tarantool and run `make test`. I don't know all those symlinks and PATH tweaks are here. Maybe they were required a long time ago (more than 4 years).

**Comment by Totktonada** [comment]
> ``dpkg-buildpackage`` or ``rpmbuild``

To be honest, I'm not sure those tools will works seamlessly. We use packpack and it is the only reliable way to build tarantool package ATM.

I would just refer [this](https://github.com/tarantool/tarantool/wiki/Build-RPM-or-Deb-package-using-packpack) instruction.

**Comment by Totktonada** [comment]
> $ # if you installed tarantool locally after build
> $ tarantool

It is either when we're about installing to some custom destdir and setting PATH (which is not actually necessary AFAIU). Or regarding installing directly to /usr or /usr/local bypassing a system package manager, which is not what I would suggest for anyone.

**Comment by Totktonada** [comment]
> * ``Release`` -- used only if the highest performance is required

We don't deploy and don't verify such packages. And the point regarding better performance comparing to `RelWithDebInfo` is dubious. So I would remove mention of the `Release` build type.

**Comment by Totktonada** [comment]
> * `OpenSSL <https://www.openssl.org>`_ library, version 1.0.1+

There are no problems with 1.1.* known to me.

**Comment by Totktonada** [comment]
> > $ # if you installed tarantool locally after build
> > $ tarantool
> 
> It is either when we're about installing to some custom destdir and setting PATH (which is not actually necessary AFAIU). Or regarding installing directly to /usr or /usr/local bypassing a system package manager, which is not what I would suggest for anyone.

However, okay, if it is /usr/local, it is more or less okay. Still requires root priveleges and dubious suggestion. Debatable topic.

**Comment by Onvember** [comment]
Outdated. See https://github.com/tarantool/doc/pull/1989

**Pull request closed without merging**